build.sbt

```diff
@@ -31,31 +34,49 @@ lazy val root = (project in file("."))
 
     libraryDependencies ++= Seq(
       "ch.qos.logback" % "logback-classic" % LogbackVersion,
-      "com.iheart" %% "ficus" % FicusVersion,
+      "com.github.pureconfig" % "pureconfig_2.12" % PureConfigVersion,
+      "com.github.tminglei" %% "slick-pg" % SlickPgVersion,
       "com.opentable.components" % "otj-pg-embedded" % OtjPgEmbeddedVersion,
       "com.typesafe" % "config" % TypesafeConfigVersion,
       "com.typesafe.scala-logging" %% "scala-logging" % TypsafeLoggingVersion,
-      "org.http4s" %% "http4s-blaze-server" % Http4sVersion,
+      "com.typesafe.slick" %% "slick" % SlickVersion,
+      "com.typesafe.slick" %% "slick-codegen" % SlickVersion,
+      "com.typesafe.slick" %% "slick-hikaricp" % SlickVersion,
+      "io.circe" %% "circe-core" % CirceVersion,
+      "io.circe" %% "circe-generic" % CirceVersion,
+      "io.circe" %% "circe-java8" % CirceVersion,
+      "io.circe" %% "circe-parser" % CirceVersion,
       "org.http4s" %% "http4s-blaze-client" % Http4sVersion,
+      "org.http4s" %% "http4s-blaze-server" % Http4sVersion,
       "org.http4s" %% "http4s-circe" % Http4sVersion,
       "org.http4s" %% "http4s-dsl" % Http4sVersion,
       "org.liquibase" % "liquibase-core" % LiquibaseVersion,
       "org.postgresql" % "postgresql" % PostgresqlVersion,
+      "org.tpolecat" %% "doobie-core" % DoobieVersion,
+      "org.tpolecat" %% "doobie-hikari" % DoobieVersion,
+      "org.tpolecat" %% "doobie-postgres" % DoobieVersion,
 
       "org.specs2" %% "specs2-core" % Specs2Version % Test,
-    ) ++ Seq(
-      "io.circe" %% "circe-core",
-      "io.circe" %% "circe-java8",
-      "io.circe" %% "circe-generic",
-      "io.circe" %% "circe-parser"
-    ).map(_ % CirceVersion),
+    ),
 
     buildInfoPackage := "com.lunatech.iamin.rest",
 
-    addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.6"),
-    addCompilerPlugin("com.olegpy" %% "better-monadic-for" % "0.2.4"),
+    addCompilerPlugin("com.olegpy" %% "better-monadic-for" % "0.3.0"),
+    addCompilerPlugin("org.spire-math" %% "kind-projector" % "0.9.10"),
 
     guardrailTasks in Compile := List(
       ScalaServer(file("documentation/api.yaml"), pkg = "com.lunatech.iamin.rest", framework = "http4s", tracing = false)
     )
   )
+
+lazy val generateSlickTables = taskKey[Seq[File]]("Generate Slick code from Liquibase migrations")
+generateSlickTables := {
+  val f = sourceDirectory.value / "main" / "scala"
+  val cp = (Compile / fullClasspath).value
+  val log = streams.value.log
+  val r = (Compile / runner).value
+
+  r.run("com.lunatech.iamin.database.SlickTableGenerator", cp.files, Array(f.getPath), log)
+
+  f.listFiles()
```

Comment:
How can we use file array in the future?

Reply:
We don't. I think sbt just needs a return value somehow. Not really relevant in our case anyway.